ols/src/tools/tools.py

```diff
@@ -1,103 +1,69 @@
 """Functions/Tools definition."""
 
 import logging
-from typing import Optional
+from typing import Callable
 
 from langchain_core.messages import ToolMessage
-from langchain_core.tools.base import BaseTool
-from pydantic import ValidationError
-
-from ols.src.tools.oc_cli import (
-    oc_adm_top,
-    oc_describe,
-    oc_get,
-    oc_logs,
-    oc_status,
-    show_pods_resource_usage,
-)
+from langchain_mcp_adapters.client import MultiServerMCPClient
 
 logger = logging.getLogger(__name__)
 
 
-oc_tools = {
-    "oc_get": oc_get,
-    "oc_describe": oc_describe,
-    "oc_logs": oc_logs,
-    "oc_adm_top": oc_adm_top,
-    "oc_status": oc_status,
-    "show_pods_resource_usage": show_pods_resource_usage,
-}
-
-
-def get_available_tools(
-    introspection_enabled: bool, user_token: Optional[str] = None
-) -> dict[str, BaseTool]:
-    """Get available tools based on introspection and user token."""
-    if not introspection_enabled:
-        logger.info("Introspection disabled; no tools available")
-        return {}
-
-    if user_token is None or (isinstance(user_token, str) and user_token == ""):
-        logger.warning("No user token provided; no oc tools available")
-        return {}
-
-    logger.info("User token provided; adding 'oc' tools")
-    return oc_tools
-
-
-def check_tool_description_length(tool: BaseTool) -> None:
-    """Check if the description of a tool is too long."""
-    if len(tool.description) >= 1024:
-        raise ValueError(f"Tool {tool.name} description too long")
-
-
-# NOTE: Limit for a tool description is 1024 characters
-for tool in oc_tools.values():
-    check_tool_description_length(tool)
-
-
-def execute_oc_tool_calls(
-    tools_map: dict,
+def get_tool_by_name(tool_name: str, mcp_client: MultiServerMCPClient) -> Callable:
```

Comment:
Given you're calling `arun` later on the result of this method, the signature should probably be `-> BaseTool`?

Reply:
It is actually `StructuredTool`. Updated.